src/iop/channelmixerrgb.c

```diff
@@ -2036,11 +2036,7 @@ void gui_changed(dt_iop_module_t *self, GtkWidget *w, void *previous)
   if(self->enabled && !(p->illuminant == DT_ILLUMINANT_PIPE || p->adaptation == DT_ADAPTATION_RGB))
   {
     // this module instance is doing chromatic adaptation
-    dt_iop_order_entry_t *CAT_instance = self->dev->proxy.chroma_adaptation;
-    dt_iop_order_entry_t *current_instance
-        = dt_ioppr_get_iop_order_entry(self->dev->iop_order_list, "channelmixerrgb", self->multi_priority);
-
-    if(CAT_instance && CAT_instance->o.iop_order != current_instance->o.iop_order)
+    if(!is_module_cat_on_pipe(self))
```

Comment:
I'm not sure to understand this one.

You check if self is not on the PIPE already... And you say in this path "another channelmixerrgb instance is doing CAT earlier in the pipe" ? Can you clarify ?

Reply:
`declare_cat_on_pipe(struct dt_iop_module_t *self, gboolean preset)` registers instances of color calibration doing CAT with a conflict resolution (in case of conflict : first one in pipe order gets the seat). So, anytime, we only need to check if the current instance is the one registered on the pipe. If not and the instance performs CAT, then we have a problem.

Reply:
I see, thanks.